script.js

```diff
@@ -11,3 +45,471 @@ function hideColumnContents(className) {
+
+      if (newColumnlineLeftIsInMinWidth) {
+        columnlineThree.style.left = windowCoordinatesX + 'px';
+      }
+      else {
+        document.removeEventListener('mousemove', onMouseMove);
+        const diffColumnlineThree = oldColumnlineThreePosition - columnlineThree.offsetLeft;
+
+        let diffIsNegative = diffColumnlineThree < 0;
+
+        if (diffIsNegative) {
+          columnTwoWidth = columnlineThree.offsetLeft - headerOne.offsetWidth;
+          columnThreeWidth = minWidth;
+          for (const element of columnTwo) {
+            element.style.width = columnTwoWidth + 'px';
+          }
+
+          headerTwo.style.width = columnTwoWidth + 'px';
+
+          for (const element of columnThree) {
+            element.style.width = columnThreeWidth + 'px';
+            
+          }
+
+          headerThree.style.width = columnThreeWidth + 'px';
+        }
+  
+        if (!diffIsNegative) {
+          columnTwoWidth = minWidth;
+          columnThreeWidth = columnThreeWidth + diffColumnlineThree;
+          for (const element of columnTwo) {
+            element.style.width = columnTwoWidth + 'px';
+            
+          }
+
+          headerTwo.style.width = columnTwoWidth + 'px';
+  
+          for (const element of columnThree) {
+            element.style.width = columnThreeWidth + 'px';
+            
+          }
+
+          headerThree.style.width = columnThreeWidth + 'px';
+        }
+        
+        for (const element of columnOne) {
+          element.style.width = columnOneWidth + 'px';
+        }
+
+        headerOne.style.width = columnOneWidth + 'px';
+  
+        for (const element of columnFour) {
+          element.style.width = columnFourWidth + 'px';
+        }
+
+        headerFour.style.width = columnFourWidth + 'px';
+
+        return;
+      }
+    }
+
+    moveToMouse(event.clientX);
+
+    function onMouseMove(event) {
+      moveToMouse(event.clientX);
+    }
+
+    document.addEventListener('mousemove', onMouseMove);
+
+    columnlineThree.onmouseup = function(event) {
+      document.removeEventListener('mousemove', onMouseMove);
+      const diffMousePosition = oldmousePosition - event.clientX;
+
+      columnTwoWidth = headerTwo.offsetWidth;
+      columnThreeWidth = headerThree.offsetWidth;
+
+      const diffIsNegative = diffMousePosition < 0;
+
+      if (diffIsNegative) {
+        columnTwoWidth = columnTwoWidth - diffMousePosition;
+        columnThreeWidth = columnThreeWidth + diffMousePosition; 
+
+        for (const element of columnTwo) {
+          element.style.width = columnTwoWidth + 'px';
+          headerTwo.style.width = columnTwoWidth + 'px';
+        }
+
+        for (const element of columnThree) {
+          element.style.width = columnThreeWidth + 'px';
+          headerThree.style.width = columnThreeWidth + 'px';
+        }
```

Comment:
Repetitive code. Same applies here. Introduce a new method with header, column and column width as parameters.